ducktape/tests/test.py

```diff
@@ -137,3 +129,168 @@ def is_test_class(self, obj):
+        header += self.pass_fail(self.results.get_aggregate_success()) + "\n"
+        header += "------------------------------------------------------------\n"
+
+        return header
+
+    def result_string(self, result):
+        result_str = ""
+        result_str += self.pass_fail(result.success) + ": " + result.test_name + "\n"
+        if not result.success:
+            result_str += "    " + result.summary + "\n"
+        if result.data is not None:
+            result_str += json.dumps(result.data) + "\n"
+
+        return result_str
+
+    def report_string(self):
+        report_str = ""
+        report_str += self.header_string()
+
+        for result in self.results:
+            report_str += self.result_string(result)
+
+        return report_str
+
+
+class SimpleFileReporter(SimpleReporter):
+    def report(self):
+        report_file = os.path.join(self.results.test_session_context.test_session_report_dir, "summary")
+        with open(report_file, "w") as fp:
+            fp.write(self.report_string())
+
+
+class SimpleStdoutReporter(SimpleReporter):
+    def report(self):
+        print self.report_string()
+
+
+class TestSessionContext(object):
+    def __init__(self, test_session_report_dir, test_session_id):
+        self.test_session_report_dir = test_session_report_dir
+        self.test_session_id = test_session_id
+
+
+class TestResult(object):
+    def __init__(self, test_session_context, test_name, success=True, summary="", data=None):
+        self.test_session_context = test_session_context
+        self.test_name = test_name
+        self.success = success
+        self.summary = summary
+        self.data = data
+
+
+class TestResults(object):
+    # TODO make this tread safe - once tests are run in parallel, this will be shared by multiple threads
+
+    def __init__(self, test_session_context):
+        # test_name -> test_result
+        self.results_map = {}
+
+        # maintains an ordering of test_results
+        self.results_list = []
+
+        # Aggregate success of all results
+        self.success = True
+
+        self.test_session_context = test_session_context
+
+    def add_result(self, test_result):
+        assert test_result.__class__ == TestResult
+        self.results_map[test_result.test_name] = test_result
+        self.results_list.append(test_result)
+        self.success = self.success and test_result.success
+
+    def get_result(self, result_name):
+        return self.results_map.get(result_name)
+
+    def get_aggregate_success(self):
+        """Check cumulative success of all tests run so far"""
+        if not self.success:
+            return False
+
+        for result in self:
+            if not result.success:
+                return False
+
+        return True
+
+    def __iter__(self):
+        for item in self.results_list:
+            yield item
+
+
+class TestRunner(Logger):
+    DEFAULT_TEST_RUN_ID_FILE = ".ducktape/test_run_id"
```

Comment:
I saw this in another file somewhere. I think this needs to be defined in one place and then all the code that uses the value needs to get the setting from the same place.